addons/account/models/account.py

```diff
@@ -669,7 +669,18 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
         # the 'Account' decimal precision + 5), and that way it's like
         # rounding after the sum of the tax amounts of each line
         prec = currency.decimal_places
-        if company_id.tax_calculation_rounding_method == 'round_globally' or not bool(self.env.context.get("round", True)):
+
+        # In some cases, it is necessary to force/prevent the rounding of the tax and the total
+        # amounts. For example, in SO/PO line, we don't want to round the price unit at the
+        # precision of the currency.
+        # The context key 'round' allows to force the standard behavior.
+        round_tax = False if company_id.tax_calculation_rounding_method == 'round_globally' else True
+        round_total = True
+        if 'round' in self.env.context:
+            round_tax = bool(self.env.context['round'])
+            round_total = bool(self.env.context['round'])
+
+        if not round_tax:
             prec += 5
```

Comment:
Ok, I understand and agree (about the scope), but then you might eventually poke the persons personally who are in charge of this as the issue seems to be in sleep mode again and it is discussed with heat again and again without useful solution until now whereas I think my concept or something similar would resolve it once and for all ;-)